mule-extensions-api/src/main/java/org/mule/runtime/extension/api/metadata/ComponentMetadataConfigurerFactoryUtils.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.metadata;
+
+import static java.util.ServiceLoader.load;
+import java.util.Iterator;
+
+/**
+ * Factory of {@link ComponentMetadataConfigurer}.
+ *
+ * @since 1.8
+ */
+public class ComponentMetadataConfigurerFactoryUtils {
+
+  /**
+   * @return an implementation of {@link ComponentMetadataConfigurer.ComponentMetadataConfigurerFactory}.
+   */
+  public ComponentMetadataConfigurer.ComponentMetadataConfigurerFactory create() {
```

Comment:
To me this class shouldn't exist. To me this should be the factory itself and use SPI to find a delegate. See the example of `org.mule.runtime.api.metadata.AbstractDataTypeBuilderFactory` , only that in this case I would make certain improvemetns:

* don't call it abstract
* don't search for an implementation of this class, but of a non api delegate
* perform the search in the context of a lazy value instead of a static block

Also, this class should be final and non-instantiable, consumed through static methods